### PR TITLE
Update `summarize_coxreg` to work with pagination in all cases

### LINKED ISSUE #952: summarize_coxreg multivar=FALSE table can not be paginated
https://github.com/insightsengineering/tern/blob/4980e3275205f63a640e391a80a166801f23b276/R/summarize_coxreg.R#L313

```r
library(scda)
library(dplyr)
library(tern)

adsl <- synthetic_cdisc_dataset("latest", "adsl")
adtte <- synthetic_cdisc_dataset("latest", "adtte")

# Ensure character variables are converted to factors and empty strings and NAs are explicit missing levels.
adsl <- df_explicit_na(adsl)
adtte <- df_explicit_na(adtte)

adsl_filtered <- adsl %>% dplyr::filter(
  RACE %in% c("ASIAN", "BLACK OR AFRICAN AMERICAN", "WHITE")
)
adtte_filtered <- dplyr::inner_join(
  x = adsl_filtered[, c("STUDYID", "USUBJID")],
  y = adtte,
  by = c("STUDYID", "USUBJID")
)

anl <- adtte_filtered %>%
  filter(PARAMCD == "OS") %>%
  mutate(EVENT = 1 - CNSR) %>%
  filter(ARM %in% c("B: Placebo", "A: Drug X")) %>%
  mutate(ARM = droplevels(relevel(ARM, "B: Placebo"))) %>%
  mutate(RACE = droplevels(RACE) %>% formatters::with_label("Race"))

# Add variable for column split label
anl <- anl %>% mutate(col_label = "Treatment Effect Adjusted for Covariate")

variables <- list(
  time = "AVAL",
  event = "EVENT",
  arm = "ARM",
  covariates = c("AGE", "SEX", "RACE")
)

lyt <- basic_table() %>%
  split_cols_by("col_label") %>%
  summarize_coxreg(variables = variables, multivar= FALSE) %>%
  append_topleft("Effect/Covariate Included in the Model")

result <- build_table(lyt = lyt, df = anl)
result

cat(export_as_txt(result, lpp = 100))
```

gives error 

Error in find_pag(pagdf, start, guess, rlpp = adjrlpp, min_siblings = min_siblings,  : 
  Unable to find any valid pagination between 1 and 6

### PR DESCRIPTION
**Pull request by edelarua**
Closes #952 

In the univariate no interaction case covariates are now printed in data rows so that pagination can be applied. No change to other cases. Tested all scenarios with pagination and all are now working.

### CODE REVIEW

File: R/summarize_coxreg.R [hunk 1]
```diff
@@ -386,15 +390,31 @@ summarize_coxreg <- function(lyt,
         varlabels = varlabels,
         split_label = "Covariate:",
         nested = FALSE,
+        child_labels = if (multivar || control$interaction || !"arm" %in% names(variables)) "default" else "hidden",
```
Comment (Melkiades) [inline R/summarize_coxreg.R:393]: Oh this is smart!

File: tests/testthat/test-summarize_coxreg.R [hunk 2]
```diff
@@ -90,10 +90,18 @@ testthat::test_that("summarize_coxreg adds the univariate Cox regression layer t
...
+      variables = variables,
+      control = control_coxreg(ties = "breslow", conf_level = 0.90)
+    ) %>%
+    build_table(df = dta_bladder)
 
   res <- testthat::expect_silent(result)
   testthat::expect_snapshot(res)
+
+  # pagination
+  testthat::expect_silent(invisible(capture.output(cat(export_as_txt(result, lpp = 10)))))
```
Comment (Melkiades) [inline tests/testthat/test-summarize_coxreg.R:104]: question: I know this is done for completeness as the pagination itself misses a couple of formatting things, but is it necessary if no wrapping or column width is inserted? In other words, is it not more intuitive to use direct pagination?

Comment (edelarua) [reply tests/testthat/test-summarize_coxreg.R]: Updated to use `paginate_table` instead.

**Review by ayogasekaram — APPROVED** [review]
lgtm @edelarua!

**Pull request merged**